src/Servant/PureScript/Settings.purs

```diff
@@ -31,26 +31,33 @@ newtype SPSettings_ params = SPSettings_ {
 
 type URLPiece = String
 
+class ToUrlPiece a where
+  toUrlPiece :: a -> URLPiece
+
+instance stringToUrlPiece :: ToUrlPiece String where
+  toUrlPiece = identity
+
+else instance genericRepToUrlPiece :: (Generic a rep, EncodeRep rep) => ToUrlPiece a where
+  toUrlPiece = stringify <<< genericEncodeJson
+
 -- | Just use the robust JSON format.
-gDefaultToURLPiece :: forall a. Generic a => a -> URLPiece
+gDefaultToURLPiece :: forall a. ToUrlPiece a => a -> URLPiece
 gDefaultToURLPiece = gDefaultEncodeHeader
 
 -- | Just use the robust JSON format.
-gDefaultEncodeHeader :: forall a. Generic a => a -> URLPiece
-gDefaultEncodeHeader v =
-  case toSpine v of
-    SString s -> s -- Special case string - just use it as is (http-api-data compatibility).
-    _ -> show <<< Aeson.encodeJson $ v
```

Comment:
why is this special case no longer needed?

Reply:
if I did not mess up this should be taken care of by this instance here:

https://github.com/eskimor/purescript-servant-support/pull/15/files#diff-7b8d38eeda45bc94e96d07a922b4631bR37

as far as I understood your code this one was used pass on strings as they are

Reply:
haha - yeah I completely missed that one. Looks good then. 'else instance' - that's a nice construct :-P